src/pkg/services/m365/api/item_pager.go

```diff
@@ -206,7 +207,10 @@ func deltaEnumerateItems[T any](
 // shared enumeration runner funcs
 // ---------------------------------------------------------------------------
 
-type addedAndRemovedHandler[T any] func(items []T) (map[string]time.Time, []string, error)
+type addedAndRemovedHandler[T any] func(
+	items []T,
+	getModTime getModTimer[T],
```

Comment:
I'm not really a fan of this design. I think a cleaner solution would be to wrap the items we're iterating through with our own struct that provides the function we want. We already do something similar with calendar events in another place in the code because we wanted `DisplayName` or some such function